token-lending/program/tests/repay.rs

```diff
@@ -28,19 +28,24 @@ const INITIAL_SOL_RESERVE_SUPPLY_LAMPORTS: u64 = 42_500 * LAMPORTS_TO_SOL;
 const INITIAL_USDC_RESERVE_SUPPLY_FRACTIONAL: u64 =
     lamports_to_usdc_fractional(INITIAL_SOL_RESERVE_SUPPLY_LAMPORTS);
 
-#[tokio::test]
-async fn test_success() {
+const OBLIGATION_LOAN: u64 = 100;
+const OBLIGATION_COLLATERAL: u64 = 90;
+const NUMBER_OF_TESTS: u64 = 2;
+struct TestReturn {
+    banks_client: BanksClient,
+    obligation: TestObligation,
+    usdc_reserve: TestReserve,
+}
+
+async fn setup() -> TestReturn {
     let mut test = ProgramTest::new(
         "spl_token_lending",
         spl_token_lending::id(),
         processor!(process_instruction),
     );
 
     // limit to track compute unit increase
-    test.set_bpf_compute_max_units(80_000);
-
-    const OBLIGATION_LOAN: u64 = 1;
-    const OBLIGATION_COLLATERAL: u64 = 500;
+    test.set_bpf_compute_max_units(NUMBER_OF_TESTS * 80_000);
```

Comment:
I don't see `NUMBER_OF_TESTS` used elsewhere, what is its purpose here?

Reply:
I noticed that this becomes a global counter even if you reset test multiple times - it's tied per-file I think, or potentially per run. So I was using it to scale up the number of maximum instructions to the number of tests you had in the file. Trying to figure out a better way to reset this each time you call "setup."

Reply:
The max units param restricts the number of compute units allowed for each transaction. It shouldn't be accumulated by number of tests. You should be able to set a new value per test if you want

Reply:
Hmm ok I will look into this

Reply:
So I did look into this, and indeed, even though each ProgramTest::new ostensibly makes a new ProgramTest object, it actually stores in memory by spl_token_lending::id() the program instruction count, so if you create it again, it doesn't reset the counter. So if I remove the number_of_tests variable, ie set it to 80,000 and just remake it each time, I run out of instructions and the tests die. However, if I just run a single test, the test is fine, because 80k for one is alright. I think this hasn't been a problem for you before because each of your files only has one test @jstarry

Reply:
Oh really? Nice find, that's a bug in the program test framework then. Do you mind writing an issue over in the [main monorepo project](https://github.com/solana-labs/solana) describing the problem?